### ISSUE TITLE
Problem of pix2pix on two different devices that shows 'nan' at the begining 

### ISSUE THREAD

**Issue by YujieXiang** [junyanz/pytorch-CycleGAN-and-pix2pix#1487]
When I use 2 different devices to run the pix2pix training part , one can smoothly finish the training part  but another leads to 'nan' in loss function since the begining as the figure shows. The environments and dataset(facades) are quite the same. 
![image](https://user-images.githubusercontent.com/111649799/192196600-52aba3bc-6242-4ba2-b22a-62888e25863f.png)

**Comment by taesungp** [comment]
Do you mean the same training runs well on one device, and it produces NaN on the other device? Or did you try to do multi-gpu training? In the case of former, yes, it's likely a GPU issue...
